File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/images2.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import CreateImageEditRequest, CreateImageVariationPostRequest, ImageGenerationRequest, ImageGenerationResponse

class Images2Api:
    """Images API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def create_image_variation(self, body: Optional[CreateImageVariationPostRequest] = None, params: Optional[Dict[str, Any]] = None) -> ImageGenerationResponse:
        """Create image variation"""
        return self._client.post(f"/ai/v3/v1/images/variations", data=body, params=params)

    def create_image(self, body: ImageGenerationRequest) -> ImageGenerationResponse:
        """Create image"""
        return self._client.post(f"/ai/v3/v1/images/generations", json=body)

    def create_image_edit(self, body: Optional[CreateImageEditRequest] = None, params: Optional[Dict[str, Any]] = None) -> ImageGenerationResponse:
        """Create image edit"""
        return self._client.post(f"/ai/v3/v1/images/edits", data=body, params=params)

File: sdkwork-ai-sdk-flutter/lib/src/api/images2.dart
import '../http/client.dart';
import '../models.dart';

class Images2Api {
  final HttpClient _client;
  
  Images2Api(this._client);

  /// Create image variation
  Future<ImageGenerationResponse?> createImageVariation(CreateImageVariationPostRequest? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/images/variations'), body: body, params: params, contentType: 'multipart/form-data');
    return response is ImageGenerationResponse ? response : null;
  }

  /// Create image
  Future<ImageGenerationResponse?> createImage(ImageGenerationRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/images/generations'), body: body, contentType: 'application/json');
    return response is ImageGenerationResponse ? response : null;
  }

  /// Create image edit
  Future<ImageGenerationResponse?> createImageEdit(CreateImageEditRequest? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/images/edits'), body: body, params: params, contentType: 'multipart/form-data');
    return response is ImageGenerationResponse ? response : null;
  }
}
